android/app/src/main/java/com/react/testapp/MainActivity.kt

```diff
@@ -51,9 +52,12 @@ class MainActivity : ReactActivity() {
     }
 
     private val startComponentActivity = { component: ComponentViewModel ->
+        didInitialNavigation = true;
         startActivity(newComponentActivityIntent(component))
     }
 
+    private var didInitialNavigation = false;
```

Comment:
This flag is going to be reset when activity is re-created (i.e. configuration changes). We need to store/restore it in the bundle if we want to preserve the value.